pbspro.spec

```diff
@@ -371,6 +371,22 @@ if [ "$1" != "1" ]; then
 	echo
 fi
 
+%posttrans %{pbs_server}
+# The %preun section of 14.x unconditially removes /etc/init.d/pbs
+# because it does not check whether the package is being removed
+# or upgraded. Make sure it exists here.
+if [ -r %{pbs_prefix}/libexec/pbs_init.d ]; then
+    install -D %{pbs_prefix}/libexec/pbs_init.d /etc/init.d/pbs
+fi
+
+%posttrans %{pbs_execution}
+# The %preun section of 14.x unconditially removes /etc/init.d/pbs
+# because it does not check whether the package is being removed
+# or upgraded. Make sure it exists here.
+if [ -r %{pbs_prefix}/libexec/pbs_init.d ]; then
+    install -D %{pbs_prefix}/libexec/pbs_init.d /etc/init.d/pbs
+fi
+
```

Comment:
For future upgrades (e.g. 18.1 to 19.1) don't we want to ensure pg_upgrade is installed? Also, there were some whitespace fixes in [PR 690](https://github.com/PBSPro/pbspro/pull/690) that are not present.

Reply:
We do not need pg_upgrade for upgrades. We are leaving it to the admin to do the upgrade, so it is his/her responsibility to install the upgrade package.
Moreover, if we add it to the requires section, the upgrade that would get installed would be the "default" one (for example in RHEL 7, it is 9.2.23) and the admin might have upgraded postgresql to 9.6. To perform the DB upgrade, we would need pg_upgrade from 9.6, which needs to be installed manually. 
I will look at the whitespace changes.